pkg/payload/precondition/precondition.go

```diff
@@ -71,25 +72,40 @@ func Summarize(errs []error, force bool) (bool, error) {
 		return false, nil
 	}
 	var msgs []string
+	var isWarning = true
 	for _, e := range errs {
 		if pferr, ok := e.(*Error); ok {
 			msgs = append(msgs, fmt.Sprintf("Precondition %q failed because of %q: %v", pferr.Name, pferr.Reason, pferr.Error()))
+			if !pferr.NonBlockingWarning {
+				isWarning = false
+			}
 			continue
 		}
 		msgs = append(msgs, e.Error())
+
 	}
 	msg := ""
 	if len(msgs) == 1 {
 		msg = msgs[0]
 	} else {
 		msg = fmt.Sprintf("Multiple precondition checks failed:\n* %s", strings.Join(msgs, "\n* "))
+
+		// when there are multiple precondition checks failed we will ignore if one of them is marked warning
+		isWarning = false
 	}
 
 	if force {
 		msg = fmt.Sprintf("Forced through blocking failures: %s", msg)
+		return !force, &payload.UpdateError{
```

Comment:
Previously we'd only get to this log line if we had blocking failures.  Now we can have either a warning or blocking issue.  I think we want something like:

```go
if force and !isWarning {
  msg = fmt.Sprintf("Forced through blocking failures: %s", msg)
  isWarning = true
}

return !isWarning, &payload.UpdateError{
  Nested:  nil,
  Reason:  "UpgradePreconditionCheckFailed",
  Message: msg,
  Name:    "PreconditionCheck",
}
```

Reply:
This works too. Somehow I thought we want to keep the code around force separate than the warning, just for readability